woocommerce/woocommerce
# Fix: Restore Button Styling for Add to Cart Buttons on Non-WooCommerce Pages

### Changes proposed in this Pull Request:





Closes #52907

This PR fixes button styling issues on product shortcodes by removing the `*` from the `.edit-post-visual-editor` selector in the `:where()` function.

#### Root cause?

After investigation, I found out that [this PR](https://github.com/woocommerce/woocommerce/pull/51153) introduced the issue. 

**Before**
```css
.woocommerce:where(body:not(.woocommerce-block-theme-has-button-styles)),
:where(body:not(.woocommerce-block-theme-has-button-styles)) .woocommerce {
```



**After**
```css
.woocommerce:where(body:not(.woocommerce-block-theme-has-button-styles)),
:where(body:not(.woocommerce-block-theme-has-button-styles)):where(:not(.edit-post-visual-editor *)) .woocommerce
```




### How to test the changes in this Pull Request:



1. Create a new post
2. Add a shortcode block containing `[products]`
3. Publish the post
3. For testing, I installed and activated "Twenty Twelve" theme.
4. Verify on Frontend that button styling is applied correctly, Here is how it looks like on "Twenty Twelve" theme:

| Before | After |
|--------|-------|
| ![Image](https://github.com/user-attachments/assets/0a521c49-7a43-4117-af54-edac4b8875cc) | ![Image](https://github.com/user-attachments/assets/8decebcb-1ac7-48cb-b48f-feface9a0f23) |

5. For Developers only: Verify that this issue is reproducible on trunk branch.



### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Fix: Button styles are missing on Add to cart button on non-WooCommerce pages
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
